charts/k8s-monitoring/charts/feature-integrations/templates/_integration_grafana_logs.tpl

```diff
@@ -19,28 +19,28 @@
         {{- $valueList := list }}
         {{- if .namespaces }}
           {{- $labelList = append $labelList "__meta_kubernetes_namespace" -}}
-          {{- $valueList = append $valueList (printf "(%s)" (join "|" .namespaces)) -}}
+          {{- $valueList = append $valueList (printf "(?:%s)" (join "|" .namespaces)) -}}
         {{- end }}
         {{- range $k, $v := .labelSelectors }}
           {{- if kindIs "slice" $v }}
             {{- $labelList = append $labelList (include "pod_label" $k) -}}
-            {{- $valueList = append $valueList (printf "(%s)" (join "|" $v)) -}}
+            {{- $valueList = append $valueList (printf "(?:%s)" (join "|" $v)) -}}
           {{- else }}
             {{- $labelList = append $labelList (include "pod_label" $k) -}}
-            {{- $valueList = append $valueList $v -}}
+            {{- $valueList = append $valueList (printf "(?:%s)" $v) -}}
           {{- end }}
         {{- end }}
 rule {
-  source_labels = {{ $labelList | sortAlpha | toJson }}
+  source_labels = {{ $labelList | toJson }}
   separator = ";"
-  regex = {{ $valueList | sortAlpha | join ";" | quote }}
+  regex = {{ $valueList | join ";" | quote }}
   target_label = "job"
   replacement = "integrations/grafana"
 }
 rule {
-  source_labels = {{ $labelList | sortAlpha | toJson }}
+  source_labels = {{ $labelList | toJson }}
   separator = ";"
-  regex = {{ $valueList | sortAlpha | join ";" | quote }}
+  regex = {{ $valueList | join ";" | quote }}
   target_label = "instance"
   replacement = {{ $instance.name | quote }}
```

Comment:
We gotta do something about predictable layout for this. We had the `sortAlpha` in there before because it wasn't deterministic and it failed autotests pretty regularly.

Maybe instead of `range $k, $v := .labelSelectors` we do something more like `range $key := (keys .labelSelectors | sort)`?